python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,640 @@
+
+    @property
+    def lines(self):
+        """
+        Contains the coordinates column, an offsets column, and a
+        mlines column. The mlines column  is optional. The mlines column stores
+        the indices of the offsets that indicate the beginning and end of each
+        MultiLineString segment. The absence of an `mlines` column indicates
+        there are no `MultiLineStrings` in the data source, only `LineString`s.
+        """
+        return self._lines
+
+    @property
+    def polygons(self):
+        """
+        Contains the coordinates column, a rings column specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        mpolygons column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+        """
+        return self._polygons
+
+    def __len__(self):
+        """
+        The numer of unique geometries stored in this GeoArrowBuffers.
+        """
+        points_length = len(self._points) if self.points is not None else 0
+        lines_length = len(self._lines) if self.lines is not None else 0
+        multipoints_length = (
+            len(self._multipoints) if self.multipoints is not None else 0
+        )
+        polygons_length = (
+            len(self._polygons) if self.polygons is not None else 0
+        )
+        return (
+            points_length + lines_length + multipoints_length + polygons_length
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoArrowBuffers.
+        """
+        result = GeoArrowBuffers(self)
+        return result
+
+    def to_host(self):
+        return GeoArrowBuffers(self, data_locale=pd)
+
+    def __repr__(self):
+        return (
+            f"{self.points}\n"
+            f"{self.multipoints}\n"
+            f"{self.lines}\n"
+            f"{self.polygons}\n"
+        )
+
+
+class CoordinateArray:
+    def __init__(self, xy, z=None, data_locale=cudf):
+        """
+        A GeoArrow column of points. The CoordinateArray stores all of the
+        points within a single data source, typically a cuspatial.GeoSeries,
+        in the format specified by GeoArrow.
+        """
+        self.data_location = data_locale
+        self.xy = xy
+        self.z = z
+
+    @property
+    def data_location(self):
+        """
+        A `Series` supporting python library, expected to be either `cudf` or
+        `pandas`. If `cudf`, GeoArrowBuffer is stored on the GPU and used in
+        cuspatial operations. If `pandas`, GeoArrowBuffer is stored on Host and
+        used for creating GeoPandas objects.
+        """
+        return self._data_location
+
+    @data_location.setter
+    def data_location(self, data_location):
+        if data_location not in [cudf, pd]:
```

Comment:
Nitpick, but no need for a list unless we're extending it:
```suggestion
        if data_location not in (cudf, pd):
```